traitsui/extras/tests/test_demo.py

```diff
@@ -0,0 +1,144 @@
+
+
+class TestDemoPathInitLocals(unittest.TestCase):
+    """ Test ``DemoPath.init_dic`` """
+
+    def test_init_dict_with_empty_directory(self):
+        with tempfile.TemporaryDirectory() as directory:
+            model = demo.DemoPath(
+                name=directory,
+            )
+            # traits api is still imported.
+            init_dic = model.init_dic
+            self.assertIsInstance(init_dic, dict)
+            self.assertIn("HasTraits", init_dic)
+
+    def test_init_dict_with_init_py(self):
+        with tempfile.TemporaryDirectory() as directory:
+            init_py = os.path.join(directory, "__init__.py")
+            with open(init_py, "w", encoding="utf-8") as f:
+                f.write("a = 1\n")
+            model = demo.DemoPath(
+                name=directory,
+            )
+            self.assertEqual(model.init_dic["a"], 1)
+            # traits api is still imported.
+            self.assertIn("HasTraits", model.init_dic)
+
+
+class TestDemoPathChildren(unittest.TestCase):
+    """ Integration test with DemoPath and its children
+    """
+
+    def test_init_dict_used_by_children(self):
+        # Test the __init__.py in a directory (if exists) is visible
+        # by the example scripts in that folder.
+        # Not sure if this is really needed, but this is an existing feature.
+        with tempfile.TemporaryDirectory() as directory:
+
+            init_content = "CONSTANT = 'HELLO'"
+            example_content = textwrap.dedent(
+                """
+                from . import CONSTANT
+                from . import HasTraits
+                """
+            )
+            subdir = os.path.join(directory, "Examples")
+            os.makedirs(subdir)
+            init_py = os.path.join(subdir, "__init__.py")
+            with open(init_py, "w", encoding="utf-8") as f:
+                f.write(init_content)
+            example_py = os.path.join(subdir, "example.py")
+            with open(example_py, "w", encoding="utf-8") as f:
+                f.write(example_content)
+
+            model = demo.DemoPath(
+                name=directory,
+                use_files=False,
+            )
+
+            # sanity check:
+            # This is one subdirectory
+            children = model.get_children()
+            self.assertEqual(len(children), 1)
+
+            # In that subdirectory, there is one Python file that is not
+            # __init__.py
+            subdir_node, = children
+            children = subdir_node.get_children()
+            self.assertEqual(len(children), 1)
+            example, = children
+            self.assertEqual(example.name, "example.py")
+
+            # This is the test objective: The __init__.py and traits api are
+            # loaded and accessible by the example script.
+            # Try running the code
+            example.run_code()
+            self.assertIn("CONSTANT", example.locals)
+            self.assertIn("HasTraits", example.locals)
```

Comment:
The test in `TestDemoPathInitLocals` and `TestDemoPathChildren` are related to `DemoPath.init_dic`, which is not  modified in this PR, so the tests are arguably unrelated.

So why I added these tests?: `DemoPath.init_dic` parses and executes the `__init__.py` too, and that looks related to the code I removed. I added these tests to ensure myself (and the reviewer) that what are removed in this PR does not affect how `__init__.py` is currently being used for the rest of the application.